### ISSUE TITLE
Inserted postal code doesn't match returned postal code

### ISSUE THREAD

**Issue by RX860** [openstreetmap/Nominatim#1178]
Hi,

I am fairly new to Nominatim and how it works, but I am a big fan. I do have a question/remark:

The postal code I insert does not match the postal code returned by Nominatim. Example:

input: ```geocode({'street': 'Hoofdstraat 15', 'postalcode': '2202ET'})```
Expected output: ```Location(15, Hoofdstraat, Noordwijk, Zuid-Holland, Nederland, 2202ET, Nederland, (52.2434229, 4.4299587, 0.0))```

Actual output: ```Location(15, Hoofdstraat, Valkenburg, Katwijk, Zuid-Holland, Nederland, 2235CA, Nederland, (52.1804168, 4.4350016, 0.0))```

This is probably because 'Hoofdstraat' is a common name in the Netherlands. I would however like an option 'must_match' in order to be sure you get the correct address location. Is this possible?

Thank you

**Comment by lonvia** [comment]
No, that's not possible. You need to recheck the results yourself, if you desire that function. Nominatim does a bit of fuzzy matching with respect to postcodes and will also return results where the postcode does not match because guessing postcodes correctly from OSM data is a bit of hit and miss. In this particular case you'll find that  Hoofdstraat 15 in Noordwijk is not mapped in OSM, so it returns the other Hoofdstraat 15 it can find.
